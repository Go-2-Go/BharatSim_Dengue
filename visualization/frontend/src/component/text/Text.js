import React from "react";

import {url} from "../../utils/url";

import useFetch from "../../hook/useFetch";

const Text = () => {
  const text = useFetch(url.HELLO);
  return <div>{text}</div>
}

export default Text;