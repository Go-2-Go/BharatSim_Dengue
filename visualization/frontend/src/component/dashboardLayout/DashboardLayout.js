import React from 'react';
import ReactGridLayout, {WidthProvider} from "react-grid-layout";

import 'react-grid-layout/css/styles.css'
import 'react-resizable/css/styles.css'
import './index.css'

import Text from "../text/Text";

const GridLayout = WidthProvider(ReactGridLayout);

const layout = [
  {i: 'a', x: 0, y: 0, w: 2, h: 2}
]

const DashboardLayout = () => {
  return (
    <GridLayout layout={layout} style={{background:'gray'}}>
      <div key="a"><Text /></div>
    </GridLayout>
  )
}

export default DashboardLayout;