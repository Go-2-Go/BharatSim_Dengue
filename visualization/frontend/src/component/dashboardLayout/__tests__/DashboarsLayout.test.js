import React from "react";
import {render} from "@testing-library/react";

import DashboardLayout from "../DashboardLayout";
import Text from "../../text/Text";
jest.mock("../../text/Text",()=>()=><>Hello text</>);

describe('<DashboardLayout />', function () {
  it('should match a snapshot for <DashboardLayout />', function () {
    const {container} = render(<DashboardLayout />);

    expect(container).toMatchSnapshot();
  });
});