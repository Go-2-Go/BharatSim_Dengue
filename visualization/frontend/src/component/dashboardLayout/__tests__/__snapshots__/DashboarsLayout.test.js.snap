// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardLayout /> should match a snapshot for <DashboardLayout /> 1`] = `
<div>
  <div
    class="react-grid-layout"
    style="height: 330px; background: gray;"
  >
    <div
      class="react-grid-item react-draggable cssTransforms react-resizable"
      style="width: -12px; height: 310px; position: absolute; transform: translate(10px,10px);"
    >
      Hello text
      <span
        class="react-resizable-handle react-resizable-handle-se"
      />
    </div>
  </div>
</div>
`;
